app/templates/scenario/scenario_step2.html

```diff
@@ -282,28 +295,64 @@ <h3>{{ group_names|get_item:group_name|title }}</h3>
             });
         } catch (error) {
             console.error('Error while saving topology. ' + error);
-            return Swal.fire('Grid Model Error',
-                'There are empty assets.\n Make sure all assets are filled in with data.', 'error');
             // re-enable save button
             save_button.classList.remove('disabled');
+            save_spinner.classList.add('d-none');
+            save_button.disabled = false;
+            return Swal.fire('Grid Model Error',
+                'There are empty assets.\n Make sure all assets are filled in with data.', 'error');
         }
+        // add feedback message to save button
+
+
+        save_button.addEventListener('click', () => {
+        save_button.disabled = true;
+        save_spinner.classList.remove('d-none');
+
+        // Simulate save delay
+        setTimeout(() => {
+            save_spinner.classList.add('d-none');
+            save_button.disabled = false;
+
+            // Show toast
+            save_toast.classList.remove('d-none');
+            save_toast.classList.add('show');
+
+            // Auto-hide
+            setTimeout(() => {
+            save_toast.classList.remove('show');
+            save_toast.classList.add('d-none');
+            }, 3000);
+        }, 1500);
+        });
     }
 </script>
 {% endblock end_body_scripts %}
 
 {% block footer %}
+<!-- Toast container (above footer) -->
+<div class="custom-toast-container">
+    <div id="saveToast"
+        class="toast align-items-center text-bg-success border-0 d-none"
+        role="alert">
+        <div class="d-flex">
+            <div class="toast-body">
+                <svg width="16" height="16" version="1.1" viewBox="0 0 100 100" fill="#1F567D" xmlns="http://www.w3.org/2000/svg">
+                    <path d="m65.691 35.762-21.68 21.668-9.6992-9.6992c-1.25-1.25-3.2891-1.25-4.5391 0s-1.25 3.2891 0 4.5391l11.969 11.969c0.62891 0.62891 1.4492 0.94141 2.2695 0.94141 0.82031 0 1.6406-0.30859 2.2695-0.94141l23.949-23.938c1.25-1.25 1.25-3.2891 0-4.5391-1.2617-1.2617-3.2891-1.2617-4.5391 0z"/>
+                    <path d="m50 10c-22.059 0-40 17.941-40 40s17.941 40 40 40 40-17.941 40-40-17.941-40-40-40zm0 73.578c-18.512 0-33.578-15.059-33.578-33.578s15.066-33.578 33.578-33.578 33.578 15.059 33.578 33.578-15.066 33.578-33.578 33.578z"/>
+                </svg>
+                <span>Saved successfully<span>
```

Comment:
```suggestion
                <span>Saved successfully</span>
```

Reply:
@josihoppe - I let you commit the suggestion, that way I make sure you don't get a problem if you read this after you already commited the compiled main.css